src/pybind/mgr/dashboard/frontend/src/app/shared/components/grafana/grafana.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component } from '@angular/core';
+import { EventEmitter } from '@angular/core';
+import { Input } from '@angular/core';
+import { OnChanges } from '@angular/core';
+import { OnInit } from '@angular/core';
+import { Output } from '@angular/core';
+
+import { DomSanitizer } from '@angular/platform-browser';
+import { SafeUrl } from '@angular/platform-browser';
+
+import { GrafanaService } from '../../../shared/api/grafana.service';
+
+import * as grafanaConst from '../../enum/grafana-template.enum';
+
+@Component({
+  selector: 'cd-grafana',
+  templateUrl: './grafana.component.html',
+  styleUrls: ['./grafana.component.scss']
+})
+export class GrafanaComponent implements OnInit, OnChanges {
+  grafanaSrc: SafeUrl;
+  url: string;
+  protocol: string;
+  host: string;
+  port: number;
+  baseUrl: any;
+  dashboardElement: string;
+  elementId: string;
+  dashboardVar: string;
+  grafanaExist = false;
+  dashboardPath: string;
+
+  @Input()
+  grafanaPath: string;
+  @Input()
+  grafanaStyle: string;
+
+  @Output()
+  fetchData = new EventEmitter();
+  panelStyle: any;
+
+  constructor(private sanitizer: DomSanitizer, private grafanaService: GrafanaService) {}
+
+  ngOnInit() {
+    this.grafanaService.getProtocol().subscribe((protocolData: any) => {
```

Comment:
You can summarize these requests like [here](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts#L164).